controller/src/beerocks/bml/internal/bml_internal.cpp

```diff
@@ -1127,6 +1127,81 @@ int bml_internal::process_cmdu_header(std::shared_ptr<beerocks_header> beerocks_
                              << " response, but no one is waiting...";
             }
         } break;
+        case beerocks_message::ACTION_BML_GET_CLIENT_LIST_RESPONSE: {
+            LOG(DEBUG) << "ACTION_BML_GET_CLIENT_LIST_RESPONSE received";
+            auto response =
+                beerocks_header->addClass<beerocks_message::cACTION_BML_GET_CLIENT_LIST_RESPONSE>();
+            if (!response) {
+                LOG(ERROR) << "addClass cACTION_BML_GET_CLIENT_LIST_RESPONSE failed";
+                return BML_RET_OP_FAILED;
+            }
+
+            bool promise_result = false;
+
+            //Signal any waiting threads
+            if (m_prmClientListGet) {
+                if (m_client_list && m_client_list_size) {
+                    *m_client_list_size = response->client_list_size();
+                    *m_client_list      = response->client_list_str();
```

Comment:
do we need here to copy the string or just point to it with the pointer as you did?

Reply:
I didn't
```
int N = 5;
auto P = &N; //Pointer to N
*P = 6; //Will change the value of the pointer, not the pointer itself
```
So `*m_client_list` will use the value of `m_client_list` (which is a string type) and run the `string::operator=` which copies the data.